src/ol/Feature.js

```diff
@@ -128,7 +128,9 @@ class Feature extends BaseObject {
    * @api
    */
   clone() {
-    const clone = new Feature(this.getProperties());
+    const clone = new Feature(
+      this.hasProperties() ? this.getProperties() : undefined
```

Comment:
We could also make it so `getProperties` is efficient without adding a new API method `hasProperties`.